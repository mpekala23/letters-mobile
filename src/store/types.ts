import { UserState } from "./User/UserTypes";
import { NotifState } from "./Notif/NotifTypes";
import { ContactState } from "./Contact/ContactTypes";

export interface AppState {
  user: UserState;
  notif: NotifState;
  contact: ContactState;
}
