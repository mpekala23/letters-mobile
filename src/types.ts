// TODO: make this typing better
// probably want a specific type for state abbrevs
export interface Facility {
  name: string;
  type: PrisonTypes;
  address: string;
  city: string;
  state: string;
  postal: string;
  phone: string;
}

export enum ProfilePicTypes {
  Topbar = 'Topbar',
  Contact = 'Contact',
  SingleContact = 'SingleContact',
}

export enum Storage {
  RememberToken = 'Ameelio-Token',
}

export enum PrisonTypes {
  State = 'State',
  Federal = 'Federal',
  Jail = 'County',
}

export enum SupportFAQTypes {
  DeleteLetter = 'DeleteLetter',
  NotArrived = 'NotArrived',
  WrongReturnAddress = 'WrongReturnAddress',
  WrongMailingAddress = 'WrongMailingAddress',
  TrackingNumber = 'TrackingNumber',
  TrackingError = 'TrackingError',
  TalkToAmeelio = 'TalkToAmeelio',
}

export enum LetterStatus {
  Draft = 'Draft',
  Created = 'Created',
  Mailed = 'Mailed',
  InTransit = 'In Transit',
  InLocalArea = 'In Local Area',
  OutForDelivery = 'Out for Delivery',
  Delivered = 'Delivered',
}

export enum LetterTypes {
  Postcard = 'postcard',
  Letter = 'letter',
}

export enum DeliveryReportTypes {
  received = 'received',
  unsure = 'unsure',
  notYetReceived = 'notYetReceived',
  haveNotAsked = 'haveNotAsked',
  haveNotReceived = 'haveNotReceived',
}

export interface Letter {
  type: LetterTypes;
  status: LetterStatus;
  isDraft: boolean;
  recipientId: number;
  content: string;
  photo?: Photo;
  letterId?: number; // TODO: Once we have more info on this field and lob, use this more
  expectedDeliveryDate?: Date;
  dateCreated?: Date;
  trackingEvents?: LetterTrackingEvent[];
}

export interface LetterTrackingEvent {
  id: number;
  name: string;
  location: ZipcodeInfo;
  date: Date;
}

export type TopbarRouteAction = {
  enabled: boolean;
  text: string;
  action: () => void | Promise<void>;
  blocking?: boolean;
};

export interface ZipcodeInfo {
  zip: string;
  city: string;
  state: string;
  lat?: number;
  long?: number;
}

export interface Photo {
  type?: 'image' | string;
  width?: number;
  height?: number;
  uri: string;
}
