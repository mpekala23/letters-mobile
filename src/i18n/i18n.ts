import i18n from 'i18n-js';

import en from './locales/en.json';
import es from './locales/es.json';

i18n.defaultLocale = 'en';
// uncomment line below for quick dev testing
// i18n.locale = 'es';
i18n.fallbacks = true;
i18n.translations = { en, es };

export default i18n;
