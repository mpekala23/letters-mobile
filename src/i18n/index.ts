import i18n from 'i18n-js';
import * as Localization from 'expo-localization';

import en from './locales/en.json';
import es from './locales/es.json';

i18n.defaultLocale = 'en';
i18n.fallbacks = true;
i18n.translations = { en, es };
i18n.locale = Localization.locale;
export default i18n;
