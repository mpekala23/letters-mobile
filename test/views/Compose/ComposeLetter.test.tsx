import React from 'react';
import { ComposeLetterScreen } from '@views';
import { render, toJSON, fireEvent } from '@testing-library/react-native';
import { LetterTypes, LetterStatus } from 'types';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import i18n from '@i18n';

const mockStore = configureStore([]);

const setup = () => {
  const navigation = { navigate: jest.fn(), addListener: jest.fn() };

  const store = mockStore({
    letter: {
      composing: {
        type: LetterTypes.Letter,
        status: LetterStatus.Draft,
        isDraft: true,
        recipientId: -1,
        recipientName: '',
        content: '',
        letterId: -1,
      },
      existing: {},
    },
    contact: {
      active: {
        firstName: 'First Name',
        lastName: 'Last',
        inmateNumber: '6',
        relationship: 'Brother',
      },
    },
  });

  const StoreProvider = ({ children }: { children: JSX.Element }) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return {
    navigation,
    store,
    ...render(<ComposeLetterScreen navigation={navigation} />, {
      wrapper: StoreProvider,
    }),
  };
};

describe('ComposeLetter screen', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should update the redux store when the message changes', () => {
    const { getByPlaceholderText, store } = setup();
    const input = getByPlaceholderText(i18n.t('Compose.placeholder'));
    fireEvent.changeText(input, 'new text');
    expect(store.getActions()[1].type).toBe('letter/set_content');
    expect(store.getActions()[1].payload).toBe('new text');
  });

  it('should update words left correctly', () => {
    const { getByPlaceholderText, getByText } = setup();
    const input = getByPlaceholderText(i18n.t('Compose.placeholder'));
    fireEvent.changeText(input, 'two words');
    expect(getByText('1998 left')).toBeDefined();
  });

  it('should display the first name', () => {
    const { getByText } = setup();
    expect(getByText('To: First Name')).toBeDefined();
  });
});
