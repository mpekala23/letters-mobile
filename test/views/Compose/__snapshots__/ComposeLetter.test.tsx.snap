// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComposeLetter screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <
    accessible={false}
    activeOpacity={1}
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "flex": 1,
              "flexDirection": "column",
              "padding": 16,
            },
            Object {
              "flex": 1,
              "paddingBottom": undefined,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "height": 60,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Poppins-Medium",
                  },
                  Object {
                    "flex": 1,
                    "fontSize": 14,
                    "marginTop": 8,
                  },
                ]
              }
            >
              To
              : 
              First Name
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Poppins-Medium-Italic",
                  },
                  Object {
                    "color": "#9A9A9A",
                    "marginTop": 8,
                    "paddingRight": 16,
                  },
                ]
              }
            >
              Autosaved
            </Text>
            <
              activeOpacity={0.7}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FF7171",
                    "borderRadius": 8,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 5,
                    "paddingHorizontal": 14,
                  },
                  Object {},
                  Object {
                    "height": 35,
                    "width": 130,
                  },
                  Object {},
                  Object {
                    "elevation": 2,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.22,
                  },
                ]
              }
              testID="clickable"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Poppins-Bold",
                    "fontSize": 14,
                    "opacity": 0,
                    "position": "absolute",
                  }
                }
              >
                Collapse
              </Text>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Poppins-Bold",
                    "fontSize": 14,
                    "opacity": 1,
                    "position": "absolute",
                  }
                }
              >
                Need Ideas?
              </Text>
            </>
          </View>
        </View>
        <View
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#F2F2F2",
              "borderRadius": 8,
              "flex": 1,
              "height": 50,
              "marginBottom": 10,
              "overflow": "hidden",
              "width": "100%",
            }
          }
          testID="parent"
        >
          <View
            testID="valid"
          />
          <View
            testID="unfocused"
          />
          <TextInput
            allowFontScaling={true}
            multiline={true}
            numberOfLines={1000}
            placeholder="Tell your loved one about something you did / saw today that reminded you of them!"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderColor": "rgba(0,0,0,0)",
                  "borderRadius": 8,
                  "borderWidth": 2,
                  "color": "black",
                  "fontSize": 16,
                  "paddingHorizontal": 10,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
                Object {
                  "color": "#282828",
                },
                Object {},
                Object {
                  "height": 50,
                },
                Object {
                  "flex": 1,
                  "fontSize": 18,
                  "paddingBottom": 208,
                  "paddingTop": 8,
                  "textAlignVertical": "top",
                },
                Object {
                  "fontFamily": "Poppins-Regular",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "height": 0,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <ScrollView
              keyboardShouldPersistTaps="always"
              nestedScrollEnabled={true}
              scrollEnabled={true}
              style={
                Object {
                  "flex": 1,
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "bottom": 0,
                "height": 200,
                "position": "absolute",
                "right": 520,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 200,
              }
            }
          >
            <View
              style={Object {}}
            >
              <
                style={
                  Array [
                    Object {
                      "height": 200,
                      "width": 200,
                    },
                    Object {
                      "alignItems": "center",
                      "borderRadius": 4,
                      "justifyContent": "center",
                      "overflow": "hidden",
                    },
                    Object {
                      "bottom": 10,
                      "left": 10,
                    },
                  ]
                }
                testID="clickable"
              >
                <View
                  testID="media placeholder"
                >
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight={200}
                    bbWidth={200}
                    fill="none"
                    focusable={false}
                    height={200}
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        Object {},
                        Object {
                          "flex": 0,
                          "height": 200,
                          "width": 200,
                        },
                      ]
                    }
                    vbHeight={200}
                    vbWidth={200}
                    width={200}
                    xml="<svg width=\\"200\\" height=\\"200\\" viewBox=\\"0 0 200 200\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<rect x=\\"2\\" y=\\"2\\" width=\\"196\\" height=\\"196\\" rx=\\"6.5\\" stroke=\\"#CCCCCC\\" stroke-width=\\"3\\"/>
<path d=\\"M98 75V120 M75 98H120\\" stroke=\\"#CCCCCC\\" stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"/>
</svg>
"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <RNSVGGroup
                      fill={null}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                    >
                      <RNSVGRect
                        height={196}
                        propList={
                          Array [
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        rx={6.5}
                        stroke={4291611852}
                        strokeWidth={3}
                        width={196}
                        x={2}
                        y={2}
                      />
                      <RNSVGPath
                        d="M98 75V120 M75 98H120"
                        propList={
                          Array [
                            "stroke",
                            "strokeWidth",
                            "strokeLinecap",
                            "strokeLinejoin",
                          ]
                        }
                        stroke={4291611852}
                        strokeLinecap={1}
                        strokeLinejoin={1}
                        strokeWidth={4}
                      />
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
              </>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "bottom": 0,
              "opacity": 0,
              "paddingVertical": 6,
              "position": "absolute",
              "width": 750,
            }
          }
        >
          <
            activeOpacity={1}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderTopColor": "#9A9A9A",
                "borderTopWidth": 0.3,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "height": "100%",
                    "justifyContent": "center",
                    "maxWidth": 120,
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                    Object {
                      "color": "#6D6D6D",
                    },
                  ]
                }
              >
                2000
                 left
              </Text>
            </View>
            <
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "height": "100%",
                    "justifyContent": "center",
                    "maxWidth": 120,
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={24}
                bbWidth={24}
                class="feather feather-image"
                fill="none"
                focusable={false}
                height={24}
                meetOrSlice={0}
                minX={0}
                minY={0}
                stroke="#6E757C"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {},
                    Object {
                      "flex": 0,
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={24}
                xml="<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"#6E757C\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"feather feather-image\\"><rect x=\\"3\\" y=\\"3\\" width=\\"18\\" height=\\"18\\" rx=\\"2\\" ry=\\"2\\"></rect><circle cx=\\"8.5\\" cy=\\"8.5\\" r=\\"1.5\\"></circle><polyline points=\\"21 15 16 10 5 21\\"></polyline></svg>"
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4285429116}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={2}
                >
                  <RNSVGRect
                    height={18}
                    rx={2}
                    ry={2}
                    width={18}
                    x={3}
                    y={3}
                  />
                  <RNSVGCircle
                    cx={8.5}
                    cy={8.5}
                    r={1.5}
                  />
                  <RNSVGPath
                    d="M21 15 16 10 5 21"
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </>
            <
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "height": "100%",
                    "justifyContent": "center",
                    "maxWidth": 120,
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={24}
                bbWidth={24}
                class="feather feather-check"
                fill="none"
                focusable={false}
                height={24}
                meetOrSlice={0}
                minX={0}
                minY={0}
                stroke="gray"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {},
                    Object {
                      "flex": 0,
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={24}
                xml="<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"gray\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"feather feather-check\\"><polyline points=\\"20 6 9 17 4 12\\"></polyline></svg>"
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={4286611584}
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth={2}
                >
                  <RNSVGPath
                    d="M20 6 9 17 4 12"
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </>
          </>
        </View>
      </View>
    </View>
  </>
</View>
`;
